### PR TITLE
Pynih fix opindex

### PR DESCRIPTION
**Pull request by atilaneves**

### CODE REVIEW

File: pynih/source/python/type.d [hunk 1]
```diff
@@ -913,25 +915,56 @@ private template PythonOpCmp(T) {
...
+                            throw new Exception("Slice steps other than 1 not supported in D: " ~ PyObject_Repr(key).to!string);
+
+                        auto dObj = self.to!T;
+                        return dObj.opSlice(start, stop).toPython;
+
+                    } else {
+                        throw new Exception(T.stringof ~ " cannot be sliced by " ~ PyObject_Repr(key).to!string);
+                    }
+
+                    assert(0, "Error in slicing " ~ T.stringof ~ " with " ~ PyObject_Repr(key).to!string);
```
Comment (John-Colvin) [inline pynih/source/python/type.d]: Why not just
```D
                } else if (pySliceCheck(key)) {
                    static if (__traits(compiles, T.init[Py_ssize_t.init .. Py_ssize_t.init])) {
                        const len = PyObject_Length(self);
                        Py_ssize_t start, stop, step;
                        const indicesRet = PySlice_GetIndices(key, len, &start, &stop, &step);

                        if (indicesRet < 0)
                            throw new Exception("Could not get slice indices for key '" ~ PyObject_Repr(key).to!string ~ "'");

                        if (step != 1)
                            throw new Exception("Slice steps other than 1 not supported in D: " ~ PyObject_Repr(key).to!string);

                        return (self.to!T)[start .. stop].toPython;
                    } else
                        throw new Exception(T.stringof ~ " cannot be sliced by " ~ PyObject_Repr(key).to!string);
                    assert(0);
```

then you would support all possible types of 1 dimensional D slicing.

Comment (atilaneves) [reply pynih/source/python/type.d]: I'm not sure I understand the suggestion.

Comment (John-Colvin) [reply pynih/source/python/type.d]: Calling opSlice doesn't necessarily give you a slice, what with there being two different ways of doing slice overloading in D.

So calling opSlice directly is the wrong approach, you should use the slice operators

Comment (atilaneves) [reply pynih/source/python/type.d]: Ah, I see. Thanks!

**Pull request merged**